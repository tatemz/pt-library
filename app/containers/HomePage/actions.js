import {
  CHECK_BOOK,
  CHECK_BOOK_FAILURE,
  CHECK_BOOK_SUCCESS,
} from './constants';

export function checkBook(book, method) {
  return {
    type: CHECK_BOOK,
    book,
    method,
  };
}

export function checkBookSuccess(book) {
  return {
    type: CHECK_BOOK_SUCCESS,
    book,
  };
}

export function checkBookFailure(book, error) {
  return {
    type: CHECK_BOOK_FAILURE,
    book,
    error,
  };
}
