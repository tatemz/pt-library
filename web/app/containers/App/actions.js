import {
  LOAD_LIBRARY,
  LOAD_LIBRARY_SUCCESS,
  LOAD_LIBRARY_FAILURE,
} from './constants';

export function loadLibrary() {
  return {
    type: LOAD_LIBRARY,
  };
}

export function loadLibrarySuccess(books) {
  return {
    type: LOAD_LIBRARY_SUCCESS,
    books,
  };
}

export function loadLibraryFailure(error) {
  return {
    type: LOAD_LIBRARY_FAILURE,
    error,
  };
}
