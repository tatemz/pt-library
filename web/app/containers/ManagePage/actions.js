import {
  ADD_BOOK,
  ADD_BOOK_FAILURE,
  ADD_BOOK_SUCCESS,
  DELETE_BOOK,
  DELETE_BOOK_FAILURE,
  DELETE_BOOK_SUCCESS,
  UPDATE_BOOK,
  UPDATE_BOOK_FAILURE,
  UPDATE_BOOK_SUCCESS,
} from './constants';

export function addBook(book) {
  return {
    type: ADD_BOOK,
    book,
  };
}

export function addBookSuccess(book) {
  return {
    type: ADD_BOOK_SUCCESS,
    book,
  };
}

export function addBookFailure(error) {
  return {
    type: ADD_BOOK_FAILURE,
    error,
  };
}

export function deleteBook(book) {
  return {
    type: DELETE_BOOK,
    book,
  };
}

export function deleteBookSuccess(book) {
  return {
    type: DELETE_BOOK_SUCCESS,
    book,
  };
}

export function deleteBookFailure(error) {
  return {
    type: DELETE_BOOK_FAILURE,
    error,
  };
}

export function updateBook(book) {
  return {
    type: UPDATE_BOOK,
    book,
  };
}

export function updateBookSuccess(book) {
  return {
    type: UPDATE_BOOK_SUCCESS,
    book,
  };
}

export function updateBookFailure(error) {
  return {
    type: UPDATE_BOOK_FAILURE,
    error,
  };
}
