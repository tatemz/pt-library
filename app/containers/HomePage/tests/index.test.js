import React from 'react';
import { render } from 'react-testing-library';
import { browserHistory } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from '../../../configureStore';
import { HomePage, mapDispatchToProps } from '../index';
import { loadLibrary } from '../actions';

describe('<HomePage />', () => {
  let store;

  const renderComponent = props =>
    render(
      <Provider store={store}>
        <HomePage {...props} />
      </Provider>,
    );

  const mockBook = {
    title: 'The Adventures of Foo Bar',
    author: 'Mr. Foo Bar III',
    isbn: '12345678',
    description:
      "Where did 'foo' and 'bar' come from? Find out in this timeless classic.",
  };

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should load library on first render', () => {
    const mockLoadLibrary = jest.fn();
    const { rerender } = renderComponent({ loadLibrary: mockLoadLibrary });
    rerender(
      <Provider store={store}>
        <HomePage books={[mockBook]} />
      </Provider>,
    );
    expect(mockLoadLibrary).toHaveBeenCalledTimes(1);
  });

  it('should handle loading state', async () => {
    const { findByText } = renderComponent({ loading: true });
    await expect(findByText('Loading...')).resolves.toBeTruthy();
  });

  it('should handle error state and output the error message', async () => {
    const error = Error('Foo bar');
    const { findByText } = renderComponent({ error });
    await expect(
      findByText('There was an error loading the library.'),
    ).resolves.toBeTruthy();
  });

  it('should render books and match the snapshot', async () => {
    const books = [mockBook];

    const {
      findByText,
      container: { firstChild },
    } = renderComponent({ books });

    await expect(findByText(books[0].title)).resolves.toBeTruthy();
    expect(firstChild).toMatchSnapshot();
  });

  describe('mapDispatchToProps', () => {
    it('should inject loadLibrary and dispatch when called', () => {
      const dispatch = jest.fn();
      const result = mapDispatchToProps(dispatch);
      result.loadLibrary();
      expect(dispatch).toHaveBeenCalledWith(loadLibrary());
    });
  });
});
