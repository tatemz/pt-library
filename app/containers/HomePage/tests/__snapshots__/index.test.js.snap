// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render books and match the snapshot 1`] = `
<ul>
  <li>
    The Adventures of Foo Bar
  </li>
</ul>
`;
